docs/guides/persistence.mdx

```diff
@@ -73,6 +73,68 @@ const atomWithAsyncStorage = (key, initialValue) => {
 }
 ```
 
+### Usage in Async actions
+
+When using an asynchronous storage, the reading of the created atoms are asynchronous too. That comes with some limitations. Before `useAtom` has been called at least once, the value is not set.
+This can cause an issue, for example, if you want to retrieve the value of your stored atom in an async action (write-only atom).
+
+Let's use the [atomWithStorage function in the utils bundle](../api/utils.mdx#atom-with-storage) for the examples, so that we have more options available.
+
+```typescript
+const storage = createJSONStorage(() => AsyncStorage)
+const userId = atomWithStorage('user-id-key', null, storage)
+const userInfo = atom({})
+const fetchUserInfo = atom(null, async (_get, set, payload: any) => {
+    const id = _get(userId) // This throws an error if "userId" is not loaded yet
+    const info = await fetch('https://jotai.org/users/' + id)
+    set(userInfo, info)
+})
+```
+
+We want to make sure the action will never fail. As always with asynchronous flow in jotai, we have 2 options: with or without **Suspense**.
+
+#### With Suspense
+
+You may want to preload atoms at root level of your app directly:
+
+```typescript
+const Preloader = () => {
+    useAtomValue(userId) // Trigger the "onMount" function that will load the data from the store
```

Comment:
```suggestion
  useAtomValue(userId) // Trigger the "onMount" function that will load the data from the store
```

Please use two spaces. (I think I should somehow recover prettier config for mdx files.)

Reply:
@dai-shi Oh sure, weird, i usually use 2 spaces. My bad